### DISCUSSION TITLE
I am unable to update of the form, one field specifically. I am using latest version and useFieldArray, and hooks in general mostly.

### DISCUSSION THREAD

**Discussion by m-ret** [react-hook-form/react-hook-form#5037] [Q&A]
**Describe the bug**

I am unable to update the state of the form when I change a value in a select element.
I am trying to use `setValue`, and also I am registering the field manually on a useEffect.

In the moment you change an option, you should see a text below the component specifying the type of the option you selected. Which is not happening. It happens until you click on the `+ SCENARIO` button.

**To Reproduce**

1. Click on the select element and choose another value different from the default one. 
2. Click on the `+ Scenario` button. 

**Codesandbox link (Required)**
Include a codesandbox will help us to investigate the issue quicker.

https://codesandbox.io/s/react-hook-form-example-forked-pmzi6?file=/src/index.js

**Expected behavior**
In the moment you change an option, you should see a text below the component specifying the type of the option you selected. Which is not happening. It happens until you click on the `+ SCENARIO` button.

**Screenshots**
https://user-images.githubusercontent.com/8395866/117090429-860f4c00-ad15-11eb-832b-c9e9b608bb40.mov

**Desktop (please complete the following information):**

- OS: macOS Big Sur 11.1
- Browser: Chrome Canary 
- Version  92.0.4490.0 (Official Build) canary (x86_64)

**Additional Context**
I already try with watch and setValue on every change of the select element.

**Comment by bluebill1049** [comment]
There are a lot of side effects in your `QuestionSelect`, and I don't think it's a bug instead of implementation detail.

if I understand correctly, you just want to select to update the view.

```ts
{field.variableRef?.type === "SHORT_TEXT" && <p>SHORT_TEXT</p>}
{field.variableRef?.type === "LONG_TEXT" && <p>LONG_TEXT</p>}
{field.variableRef?.type === "OTHER" && <p>OTHER</p>}
```

I don't think you need the above. 

-----

the simple solution can be something like the following:

https://codesandbox.io/s/react-hook-form-usefieldarray-forked-gdtlx?file=/src/index.js

**Reply by m-ret** [reply]
What I have to show here 
```
{field.variableRef?.type === "SHORT_TEXT" && <p>SHORT_TEXT</p>}
{field.variableRef?.type === "LONG_TEXT" && <p>LONG_TEXT</p>}
{field.variableRef?.type === "OTHER" && <p>OTHER</p>}
```
Is a completely different value from what I choose in the QuestionSelect element.
If I choose SHORT_TEXT, I should show an input text. if I choose LONG_TEXT, I should show a textarea, if I choose other, I should show a dropdown and so son.

**Reply by bluebill1049** [reply]
Did u see my codesandbox?
